### PR TITLE
Fix nonumerics in for-loops

### PR DESCRIPTION
**Pull request by thisac**
Using bool or string in for loop iterators isn't working properly. Currently, it returns None, instead of the wanted string or bool. This PR fixes the issue, so that e.g. the following works.

```# a test blackbird script
name test_name
version 1.0
target fock (num_subsystems=1, cutoff_dim=7, shots=10)

for bool i in [True, False]
    UnaryGate(i, {param}) | 0
```

### CODE REVIEW

File: blackbird_python/blackbird/tests/test_listener.py [hunk 1]
```diff
@@ -832,6 +832,24 @@ def test_for_loop(self, modes, parse_input_mocked_metadata):
...
+        )
+        
+    def test_for_str(self, parse_input_mocked_metadata):
+        """Test that a for-loop over a list containing strings is parsed correctly"""
+        bb = parse_input_mocked_metadata(
+            'for str s in ["one", "two"]\n\tMeasureFock() | 0'
+        )
+        assert np.all(
+            bb._forvar["s"] == np.array(["one", "two"])
+        )
```
Comment (josh146) [inline blackbird_python/blackbird/tests/test_listener.py:852]: Nice tests, although, should they also be testing for the case where the boolean for loop variable is used within the loop?

**Review by josh146 — APPROVED** [review]
Looks good from my end @thisac! Only suggestion is to add another test, using the boolean for loop variable within the for loop.

File: blackbird_python/blackbird/tests/test_listener.py (hunk 1 above)
Comment (thisac) [reply blackbird_python/blackbird/tests/test_listener.py]: Updated the test to use the bool inside the loop. Not sure if there's another way of testing it. 🤔

**Review by josh146 — APPROVED** [review]
Thanks Theo!

**Pull request merged**